### PR TITLE
CACTUS-440: Tab/TabList and supporting components

### PR DESCRIPTION
**Pull request by wilysword**
https://repayonline.atlassian.net/browse/CACTUS-440

It's similar enough to MenuBar that I was able to reuse some of the code with minimal refactoring; in the long run we could probably combine even more, but that was more refactoring than I wanted in a new feature. I also fixed a long-standing issue with Flex where it rendered `display="flex"` as an HTML attribute (that's what most of the updated snapshots are).

I added the `as` prop on `Tab` more for the sake of completeness than because I expect it to be used, but for some reason trying `as="button"` wouldn't display correctly on IE. There were other issues with alignment if you try to do multi-line text on a tab; and in fact, multi-line text is itself something of an issue: there's no way to combine flex with scroll & auto-soft-line-breaks, because the flex will _always_ shrink rather than overflow given the choice. Hopefully no one will need tab titles that are long enough to make that an issue.

### CODE REVIEW

**Review by Dhalton — APPROVED** [review]
LGTM, really like the `fillGaps` option.

**Review by NicolasSimmonds — APPROVED** [review]

**Review by mikesoltow — CHANGES_REQUESTED** [review]
It looks like IE11 won't let you hit ENTER when focused on a Tab to select it.  It has something to do with the element type, as if I changed `StyledTab` to `styled.button`, then it worked.  But I'm not sure if there are other side effects from making that change.

File: modules/cactus-web/src/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    flex-direction: column;
+  }
+`
+
+const StyledTab = styled.div`
+  display: block;
+  position: relative;
+  padding: 8px 16px;
+
+  &:focus::after {
```
[the remark this answers is not in the document]
Comment (wilysword) [reply modules/cactus-web/src/Tabs/Tabs.tsx:293]: I removed the focus indicator for browsers that support it, but that's not many.

**Comment by wilysword** [comment]
The enter thing is a bug in IE according to a comment in [this question](https://stackoverflow.com/questions/48288630/ie-keypress-event-not-firing-for-enter-key), but the link is dead and I can't find the original issue. It baffles me that they could have such a bug and never notice/fix it.

File: modules/cactus-web/src/Tabs/Tabs.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikesoltow) [reply modules/cactus-web/src/Tabs/Tabs.tsx:293]: Interesting...MDN says that Firefox supports it, but it doesn't seem to have any effect there.  I wonder if Firefox just has different criteria for triggering the selector.

I think I'll make a ticket to replace this implementation with a polyfill (i.e. https://github.com/WICG/focus-visible) that will be more cross-browser compatible.  But this is fine for now.

[the remark this answers is not in the document]
Comment (wilysword) [reply modules/cactus-web/src/Tabs/Tabs.tsx:293]: The version of Firefox that supports it is beta, according to Browserstack.

**Review by mikesoltow — APPROVED** [review]

File: modules/cactus-web/src/Tabs/Tabs.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikesoltow) [reply modules/cactus-web/src/Tabs/Tabs.tsx:293]: Ah, right you are.  I could go either way on the polyfill then.  I'm not fussed about IE11 support...and perhaps not having it on Safari is NBD as well.  I don't have the ticket prioritized at this point...and we can talk more about it if/when we groom it.

**Pull request merged**